src/Combinatorics/EnumerativeCombinatorics/partitions.jl

```diff
@@ -182,54 +188,55 @@ julia> collect(partitions(Int8(4))) # using less memory
  Int8[1, 1, 1, 1]
 ```
 """
-function partitions(n::T) where T <: IntegerUnion
-
-  #Argument checking
+function partitions(n::T) where {T <: IntegerUnion}
   @req n >= 0 "n >= 0 required"
+  return PartitionSet{T}(n)
+end
 
-  # Some trivial cases
-  if n == 0
-    return (p for p in Partition{T}[ partition(T[], check = false) ])
-  elseif n == 1
-    return (p for p in Partition{T}[ partition(T[1], check = false) ])
+
+function Base.iterate(P::PartitionSet{T}) where T
+  if P.n == 0
+    return Partition{T}[ partition(T[]) ], (T[], 0, 0)
+  elseif P.n == 1
+    return Partition{T}[ partition(T[1]) ], (T[1], 1, 0)
```

Comment:
```suggestion
    return partition(T[]), (T[], 0, 0)
  elseif P.n == 1
    return partition(T[1]), (T[1], 1, 0)
```

Reply:
thanks, I fixed this the same way before seeing your remark once I saw the failing tests